### ISSUE TITLE
Acceptance tests not working locally since the removal of patchwork/utf8

### ISSUE THREAD

**Issue by JammingBen** [owncloud/core#38303]
Since the removal of `patchwork/utf8` (https://github.com/owncloud/core/pull/38286), acceptance tests wont run on my local machine. It basically renders a HTML page to my terminal, including the following error:

`Setting locale to en_US.UTF-8/fr_FR.UTF-8/es_ES.UTF-8/de_DE.UTF-8/ru_RU.UTF-8/pt_BR.UTF-8/it_IT.UTF-8/ja_JP.UTF-8/zh_CN.UTF-8 failed<br>Please install one of these locales on your system and restart your webserver.`

Could be because of the missing intl fallback which the test suite needed?

cc @JanAckermann @phil-davis can you confirm this?

**Comment by phil-davis** [comment]
An API test:
```
$ make test-acceptance-api BEHAT_FEATURE=tests/acceptance/features/apiComments/comments.feature:9
composer install
Installing dependencies from lock file (including require-dev)
Verifying lock file contents can be installed on current platform.
Nothing to install, update or remove
...
Running apiComments tests tagged ~@skipWhenTestingRemoteSystems&&~@skipOnFedOcV10&&~@skipOnFedOcV10.6&&~@skipOnFedOcV10.6.1&&~@skipOnOcV10&&~@skipOnOcV10.6&&~@skipOnOcV10.6.1&&@api&&~@skip 
@api @comments-app-required @issue-ocis-reva-38
Feature: Comments

  Background:                                                                    # /home/phil/git/owncloud/core/tests/acceptance/features/apiComments/comments.feature:4
    Given using new DAV path                                                     # FeatureContext::usingOldOrNewDavPath()
    And user "Alice" has been created with default attributes and skeleton files # FeatureContext::userHasBeenCreatedWithDefaultAttributes()

  @smokeTest
  Scenario: Getting info of comments using files endpoint                                                 # /home/phil/git/owncloud/core/tests/acceptance/features/apiComments/comments.feature:9
    And user "Alice" has uploaded file "filesForUpload/textfile.txt" to "/myFileToComment.txt"            # FeatureContext::userHasUploadedAFileTo()
    And user "Alice" has commented with content "My first comment" on file "/myFileToComment.txt"         # CommentsContext::userHasCommentedWithContentOnEntry()
    And user "Alice" should have the following comments on file "/myFileToComment.txt"                    # CommentsContext::checkComments()
      | user  | comment          |
      | Alice | My first comment |
    When user "Alice" gets the following properties of folder "/myFileToComment.txt" using the WebDAV API # WebDavPropertiesContext::userGetsPropertiesOfFolder()
      | propertyName       |
      | oc:comments-href   |
      | oc:comments-count  |
      | oc:comments-unread |
    Then the single response should contain a property "oc:comments-count" with value "1"                 # WebDavPropertiesContext::theSingleResponseShouldContainAPropertyWithValue()
    And the single response should contain a property "oc:comments-unread" with value "0"                 # WebDavPropertiesContext::theSingleResponseShouldContainAPropertyWithValue()
    And the single response should contain a property "oc:comments-href" with value "%a_comment_url%"     # WebDavPropertiesContext::theSingleResponseShouldContainAPropertyWithValue()

1 scenario (1 passed)
9 steps (9 passed)
0m15.90s (17.05Mb)
runsh: Total 1 scenarios (1 passed, 0 failed)
runsh: Exit code of main run: 0
runsh: There were no unexpected failures.
runsh: There were no unexpected success.
```

and UI test:
```

```
$ make test-acceptance-webui BEHAT_FEATURE=tests/acceptance/features/webUIComments/comments.feature:17
composer install
Installing dependencies from lock file (including require-dev)
Verifying lock file contents can be installed on current platform.
Nothing to install, update or remove
...
Running webUIComments tests tagged ~@skipWhenTestingRemoteSystems&&~@skipOnFedOcV10&&~@skipOnFedOcV10.6&&~@skipOnFedOcV10.6.1&&~@skipOnOcV10&&~@skipOnOcV10.6&&~@skipOnOcV10.6.1&&@webUI&&~@skip on browser 'chrome' 
@webUI @insulated @disablePreviews @comments-app-required
Feature: Add, delete and edit comments in files and folders
  
  As a user
  I would like to add, delete and edit comments in any file/folder
  So that I can provide more information about the file/folder

  Background:                                                                               # /home/phil/git/owncloud/core/tests/acceptance/features/webUIComments/comments.feature:8
    Given these users have been created with default attributes and without skeleton files: # FeatureContext::theseUsersHaveBeenCreatedWithDefaultAttributesAndWithoutSkeletonFiles()
      | username |
      | Alice    |
      | Brian    |
    And user "Alice" has uploaded file with content "does-not-matter" to "/lorem.txt"       # FeatureContext::userHasUploadedAFileWithContentTo()
    And user "Alice" has logged in using the webUI                                          # WebUILoginContext::theUserHasLoggedInUsingTheWebUI()
    And the user has browsed to the files page                                              # WebUIFilesContext::theUserBrowsesToTheFilesPage()

SCENARIO RESULT: (pass)
  Scenario Outline: user adds and deletes comment for a file/folder                                    # /home/phil/git/owncloud/core/tests/acceptance/features/webUIComments/comments.feature:17
    When the user browses directly to display the "comments" details of file "lorem.txt" in folder "/" # WebUIFilesContext::theUserBrowsesDirectlyToDetailsTabOfFileInFolder()
    And the user comments with content "<comment>" using the webUI                                     # WebUIFilesContext::theUserCommentsWithContentUsingTheWebui()
    Then the comment "<comment>" should be listed in the comments tab in details dialog                # WebUIFilesContext::theCommentShouldBeListedInTheCommentsTabInDetailsDialog()
    When the user deletes the comment "<comment>" using the webUI                                      # WebUIFilesContext::theUserDeletesTheCommentUsingTheWebui()
    Then the comment "<comment>" should not be listed in the comments tab in details dialog            # WebUIFilesContext::theCommentShouldBeListedInTheCommentsTabInDetailsDialog()

    Examples:
      | comment     |
      | lorem ipsum |
SCENARIO RESULT: (pass)
      | 😀 🤖         |
SCENARIO RESULT: (pass)
      | नेपालि      |

3 scenarios (3 passed)
27 steps (27 passed)
2m11.11s (25.59Mb)
runsh: Total 3 scenarios (3 passed, 0 failed)
runsh: Exit code of main run: 0
runsh: There were no unexpected failures.
runsh: There were no unexpected success.
```

It is working for me. But I run a PHP dev server locally. I don't bother putting my core git clone inside an Apache server. So YMMV.

**Comment by phil-davis** [comment]
And I have `php-intl` (and lots of other) extensions installed.

**Comment by JanAckermann** [comment]
this is weird, so the intl plugin is installed on your env @JammingBen ? can you revert to the before state and try it again?

**Comment by JammingBen** [comment]
> It is working for me. But I run a PHP dev server locally. I don't bother putting my core git clone inside an Apache server. So YMMV.

Same as I do.

> this is weird, so the intl plugin is installed on your env @JammingBen ? can you revert to the before state and try it again?

Yes, I also have no problems with accessing the webui itself. Tests keeps failing with the current master. When reverting the changes from https://github.com/owncloud/core/pull/38286, tests run fine.

**Comment by ho4ho** [comment]
Related / duplicate by #38263 ? Some analysis by @created4el  by https://github.com/owncloud/core/issues/38263#issuecomment-753521360

**Comment by JanAckermann** [comment]
@JammingBen the issue @ho4ho mentioned is quite before the patchwork/utf8 package was removed, the checks are still the same, so if php can't load the locale packages (from system) it throws this kind of error.
Not sure if it relates due remove of the patchwork/utf8 package

**Comment by created4el** [comment]
Interestingly, I just upgraded my debian to 10.4 because I needed to update something else and so I tried to see if the setLocale(LC_CTYPE,0) shows before and after invoking initLocale() and to see if the if (\basename('ss') === '') was true... and apparently basename is now no longer '' so the check doesn't kick out due to mis-identifying whether initLocale() actually worked or not. And the setLocale(LC_CTYPE,0) returns US.UTF-8 both before and after the initLocale(). I other words, I can no longer demonstrate that the check doesn't actually work as it seems it is intended.

At any rate, this whole issue could have been avoided if setLocale(LC_CTYPE,0) were used as the check instead of examining basename which apparently changed its behavior... unless you actually need basename('ss') to be set to something which seems odd... but I don't know your system that well.
